Integrations/integration-Thinkst_Canary.yml

```diff
@@ -0,0 +1,558 @@
+        res_json = yaml.safe_load(res.text)
+        new_devices = [
+            {new_key: device[old_key] if old_key in device else None for old_key, new_key in RELEVANT_DEVICE_ENTRIES.items()} for
+            device in res_json['devices']]
+        return res_json, new_devices
+
+
+    def list_canaries_command():
+        res_json, new_devices = list_canaries()
+        demisto.results({
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['json'],
+            'Contents': res_json,
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown('Canary Devices', new_devices,
+                                             headers=['ID', 'Name', 'Description', 'Address', 'Status', 'Location', 'Version',
+                                                      'LastSeen', 'LastUpdated']),
+            'EntryContext': {'CanaryTools.Device(val.ID && val.ID == obj.ID)': new_devices}
+        })
+
+
+    def list_tokens():
+        # Retrieve all Canary Tokens available in Canary Tools
+        params = {
+            'auth_token': AUTH_TOKEN
+        }
+
+        res = session.request('GET', SERVER + 'canarytokens/fetch', params=params, verify=not UNSECURED)
+        res_json = yaml.safe_load(res.text)
+        new_tokens = [{new_key: token[old_key] if old_key in token else None for old_key, new_key in RELEVANT_TOKEN_ENTRIES.items()}
+                      for token in res_json['tokens'][:-1]]
+        return res_json, new_tokens
+
+
+    def list_tokens_command():
+        res_json, new_tokens = list_tokens()
+        demisto.results({
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['json'],
+            'Contents': res_json,
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown('Canary Tools Tokens', new_tokens, headers=sorted(new_tokens[0].keys())),
+            'EntryContext': {'CanaryTools.Tokens(val.CanaryToken && val.CanaryToken == obj.CanaryToken)': new_tokens}
+        })
+
+
+    def get_token_command():
+        # Fetch a Canary Token from the Canary Tools server
+        token = demisto.args().get('token')
+        params = {
+            'auth_token': AUTH_TOKEN,
+            'canarytoken': token
+        }
+
+        res = session.request('GET', SERVER + 'canarytoken/fetch', params=params, verify=not UNSECURED).json()
+        results = {
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['json'],
+            'Contents': res,
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': 'File Fetched Successfully!',
+            'EntryContext': {
+                'CanaryTools.Tokens(val.CanaryToken && val.CanaryToken == obj.CanaryToken)': res.get('token').get('canarytoken')}
+        }
+
+        if res.get('token').get('doc'):
+            name = res.get('token').get('doc_name')
```

Comment:
you should `{}` to be the default of the first `.get()`

Reply:
if you're talking about `if res.get('token').get('doc'):` then the default return would be `None` in case there's no result, so it will not go inside the condition, as it's supposed to.

Reply:
i meant that it if `res.get('token')` return `None` then an AttributeError would raise since NoneType object got no attribute `get`